src/Reflection/ReflectionParameter.php

```diff
@@ -388,12 +401,17 @@ public function getPosition() : int
      */
     public function getTypeHint() : ?Type
     {
+        return $this->getRealClassName($this->node->type);
+    }
+
+    private function getRealClassName($className) : ?Type
```

Comment:
If there are 3 possible types, then a docblock is OK, although that's a bit of a smell.